### PR TITLE
Implement submission (EXPOSUREAPP-8954)

### PR DESCRIPTION
**Pull request by mtwalli**
- Create organiser submission Repository
- Added Playbook logic for organiser submission 
- Prepare check-ins for submissions

# TODO in a follow up PR 
~~After API task -> wire submission to API endpoints~~
- Error handling 
- Unit Tests

### CODE REVIEW

File: Corona-Warn-App/src/main/java/de/rki/coronawarnapp/playbook/OrganizerPlaybook.kt [hunk 1]
```diff
@@ -0,0 +1,159 @@
...
+import kotlinx.coroutines.CoroutineScope
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.delay
+import kotlinx.coroutines.launch
+import timber.log.Timber
+import java.util.UUID
+import java.util.concurrent.TimeUnit
+import javax.inject.Inject
+
+class OrganizerPlaybook @Inject constructor(
```
Comment (BMItr) [inline Corona-Warn-App/src/main/java/de/rki/coronawarnapp/playbook/OrganizerPlaybook.kt:19]: Simply inject and use AppScope

File: Corona-Warn-App/src/main/java/de/rki/coronawarnapp/presencetracing/organizer/submission/server/OrganizerSubmissionServer.kt [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+import de.rki.coronawarnapp.appconfig.AppConfigProvider
+import de.rki.coronawarnapp.presencetracing.checkins.CheckInsReport
+import de.rki.coronawarnapp.server.protocols.internal.SubmissionPayloadOuterClass.SubmissionPayload
+import de.rki.coronawarnapp.util.PaddingTool
+import de.rki.coronawarnapp.util.coroutine.DispatcherProvider
+import kotlinx.coroutines.withContext
+import timber.log.Timber
+import javax.inject.Inject
+
+class OrganizerSubmissionServer @Inject constructor(
```
Comment (BMItr) [inline Corona-Warn-App/src/main/java/de/rki/coronawarnapp/presencetracing/organizer/submission/server/OrganizerSubmissionServer.kt:13]: @Reuseable or @Singleton?

File: Corona-Warn-App/src/main/java/de/rki/coronawarnapp/presencetracing/organizer/submission/server/OrganizerSubmissionServer.kt [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+            .addAllVisitedCountries(emptyList())
+            .addAllCheckIns(checkInsReport.unencryptedCheckIns)
+            .addAllCheckInProtectedReports(checkInsReport.encryptedCheckIns)
+            .setSubmissionType(SubmissionPayload.SubmissionType.SUBMISSION_TYPE_HOST_WARNING)
+            .build()
+
+        // TODO submit
+    }
+
+    suspend fun submitFakePayload() = withContext(dispatcherProvider.IO) {
```
Comment (BMItr) [inline Corona-Warn-App/src/main/java/de/rki/coronawarnapp/presencetracing/organizer/submission/server/OrganizerSubmissionServer.kt:46]: Just asking for my PR:
Do you need some special header for fake submissions like cwa-fake?

**Comment by BMItr** [comment]
👌 looks very good. small comments & one question

File: Corona-Warn-App/src/main/java/de/rki/coronawarnapp/presencetracing/organizer/submission/server/OrganizerSubmissionServer.kt (hunk 3 above)
Comment (mtwalli) [reply Corona-Warn-App/src/main/java/de/rki/coronawarnapp/presencetracing/organizer/submission/server/OrganizerSubmissionServer.kt:46]: Good question : @mlenkeit  what do you think? do we need a special header here ? or do empty payload is enough?

File: Corona-Warn-App/src/main/java/de/rki/coronawarnapp/playbook/OrganizerPlaybook.kt (hunk 1 above)
Comment (mtwalli) [reply Corona-Warn-App/src/main/java/de/rki/coronawarnapp/playbook/OrganizerPlaybook.kt:21]: done

File: Corona-Warn-App/src/main/java/de/rki/coronawarnapp/presencetracing/organizer/submission/server/OrganizerSubmissionServer.kt (hunk 2 above)
Comment (mtwalli) [reply Corona-Warn-App/src/main/java/de/rki/coronawarnapp/presencetracing/organizer/submission/server/OrganizerSubmissionServer.kt:15]: Done

File: Corona-Warn-App/src/main/java/de/rki/coronawarnapp/presencetracing/organizer/submission/server/OrganizerSubmissionServer.kt (hunk 3 above)
Comment (mlenkeit) [reply Corona-Warn-App/src/main/java/de/rki/coronawarnapp/presencetracing/organizer/submission/server/OrganizerSubmissionServer.kt:48]: I would suggest to send the fake submission exactly the same way like for the regular submission, even to the same server endpoint (`/diagnosis-keys`) and with `cwa-fake`

Comment (mtwalli) [reply Corona-Warn-App/src/main/java/de/rki/coronawarnapp/presencetracing/organizer/submission/server/OrganizerSubmissionServer.kt]: Done

**Review by BMItr — APPROVED** [review]

**Review by axelherbstreith — APPROVED** [review]
Lgtm

**Pull request merged**